### PR TITLE
[PDI-17726] Add missing libformula functions to formula step UI inclu…

### PR DESCRIPTION
**Pull request by cdeptula**
…ding minute and second

### CODE REVIEW

**Comment by usbrandon** [comment]
Hi Chris.  Thank you for this fix. The libformula is very useful.

**Comment by buildguy** [comment]
### :gear: Build finished in 19m 31s with command:
```
mvn clean verify -B -e -f pom.xml -Daether.connector.resumeDownloads=false -amd -Daudit -P !assemblies -U -Dmaven-surefire-plugin.argLine=-XX:MaxRAMPercentage=90.0 -Dmaven-failsafe-plugin.argLine=-XX:MaxRAMPercentage=90.0 -pl engine
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|3532|0|7|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/5/testReport)|
|----|

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 19m 10s with command:
```
mvn clean verify -B -e -f pom.xml -Daether.connector.resumeDownloads=false -amd -Daudit -P !assemblies -U -Dmaven-surefire-plugin.argLine=-XX:MaxRAMPercentage=90.0 -Dmaven-failsafe-plugin.argLine=-XX:MaxRAMPercentage=90.0 -pl engine
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|3532|0|7|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/6/testReport)|
|----|

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 18m 57s with command:
```
mvn clean verify -B -e -f pom.xml -Daether.connector.resumeDownloads=false -amd -Daudit -P !assemblies -U -Dmaven-surefire-plugin.argLine=-XX:MaxRAMPercentage=90.0 -Dmaven-failsafe-plugin.argLine=-XX:MaxRAMPercentage=90.0 -pl engine
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|3532|0|7|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/7/testReport)|
|----|

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 20m 49s with command:
```
mvn clean verify -B -e -f pom.xml -Daether.connector.resumeDownloads=false -amd -Daudit -P !assemblies -U -Dmaven-surefire-plugin.argLine=-XX:MaxRAMPercentage=90.0 -Dmaven-failsafe-plugin.argLine=-XX:MaxRAMPercentage=90.0 -pl engine
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|3530|0|7|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/8/testReport)|
|----|

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 23m 48s with command:
```
mvn clean verify -f pom.xml -Daether.connector.resumeDownloads=false -amd -Daudit -P !assemblies -U -Dmaven-surefire-plugin.argLine=-XX:MaxRAMPercentage=90.0 -Dmaven-failsafe-plugin.argLine=-XX:MaxRAMPercentage=90.0 -pl engine
```
### :warning: Failed Tests
---
#### :cloud_with_lightning_and_rain: 11 test(s) failed:
- **org.pentaho.di.trans.StepWithMappingMetaTest.loadMappingMeta**
```
    javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```
- **org.pentaho.di.trans.StepWithMappingMetaTest.loadMappingMetaTest**
```
    javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```
- **org.pentaho.di.trans.StepWithMappingMetaTest.testFileNameAsVariable**
```
    javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```
- **org.pentaho.di.trans.TransMetaTest.testLoadXml**
```
    javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```
- **org.pentaho.di.trans.TransMetaTest.testSetInternalEntryCurrentDirectoryWithRepository**
```
    javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```
- **org.pentaho.di.trans.TransMetaTest.testTransWithOneStepIsConsideredUsed**
```
    javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```
- **org.pentaho.di.www.ExecuteJobServletTest.testExecuteJobServletTest**
```
    javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```
- **org.pentaho.di.www.ExecuteJobServletTest.testExecuteJobServletTestCantFindDirectory**
```
    javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```
- **org.pentaho.di.www.ExecuteJobServletTest.testExecuteJobServletTestJobNotFoundInDirectory**
```
    javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```
- **org.pentaho.di.www.ExecuteJobServletTest.testExecuteJobServletTestCantFindRepository**
```
    javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```
- **org.pentaho.di.www.ExecuteJobServletTest.testExecuteJobServletTestWithUnauthorizedUser**
```
    javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```
  ---

|Total|Failed|Skipped|
|---|---|---|
|3539|11|7|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/10/testReport)|
|----|

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 21m 6s with command:
```
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd -Daudit -P !assemblies -pl engine
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|3539|0|7|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/11/testReport)|
|----|

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 31m 23s with command:
```
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd -Daudit -P !assemblies -pl engine
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|3539|0|7|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/12/testReport)|
|----|

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 22m 45s with command:
```
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd -Daudit -P !assemblies -pl engine
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|3539|0|7|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/13/testReport)|
|----|

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 21m 45s with command:
```
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd -Daudit -P !assemblies -pl engine
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|3538|0|7|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/14/testReport)|
|----|

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 21m 29s with command:
```
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd -Daudit -P !assemblies -pl engine
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|3540|0|7|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/15/testReport)|
|----|

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 21m 19s with command:
```
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd -Daudit -P !assemblies -pl engine
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|3541|0|7|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/16/testReport)|
|----|

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 21m 14s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -P !assemblies -pl engine
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|3544|0|7|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/18/testReport)|
|----|


:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 22m 18s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -P !assemblies -pl engine
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|3557|0|7|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/20/testReport)|
|----|


:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 22m 40s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -P !assemblies -pl engine
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|3557|0|7|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/21/testReport)|
|----|


:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 2m 27s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -P !assemblies -pl engine
```
:exclamation: No tests found!


:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 21m 59s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -P !assemblies -pl engine
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|3557|0|7|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/23/testReport)|
|----|


:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 23m 22s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -P !assemblies -pl engine
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|3557|0|7|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/24/testReport)|
|----|


:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 23m 53s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -pl engine
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|3571|0|7|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/26/testReport)|
|----|


:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 23m 55s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -pl engine
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|3572|0|7|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/27/testReport)|
|----|


:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 23m 13s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -pl engine
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|3572|0|7|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/28/testReport)|
|----|


:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 24m 34s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -pl engine
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|3579|0|7|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/30/testReport)|
|----|


:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 24m 25s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -pl engine
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|3579|0|7|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/31/testReport)|
|----|


:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 22m 20s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -pl engine
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|3579|0|7|

|[Details](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/32/testReport)|
|----|


:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 8m 51s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
:ok_hand: All tests passed!
 
Tests run: **55**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/34/testReport)

 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 24m 27s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
:ok_hand: All tests passed!
 
Tests run: **3579**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/35/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 23m 11s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
:ok_hand: All tests passed!
 
Tests run: **3588**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/36/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 22m 43s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
:ok_hand: All tests passed!
 
Tests run: **3601**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/37/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: engine/src/main/resources/org/pentaho/libformula/editor/function/functions.xml [hunk 1]
```diff
@@ -987,5 +1584,198 @@ Note: The multi-argument form is different from an "exclusive disjunction" opera
...
+    <returns>Logical</returns>
+    <constraints>None</constraints>
+    <semantics>String is the string to search, searchString is the sub string to search for</semantics>
+    <examples>
+      <example>  <expression>CONTAINS("ABC";"B")</expression>     <result>True</result>    <level>1</level>   <comment></comment>     </example>
+    </examples>
+  </function>
+  <function>
+    <category>%Category.Database</category>
+    <name>ENSWITH</name>
```
Comment (smmribeiro) [inline engine/src/main/resources/org/pentaho/libformula/editor/function/functions.xml]: Typo: should be "ENDSWITH", n ot "ENSWITH".

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 24m 17s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
:ok_hand: All tests passed!
 
Tests run: **3601**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/38/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by cdeptula** [comment]
The typo is fixed now.

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 17m 31s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
:ok_hand: All tests passed!
 
Tests run: **3601**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/39/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 25m 28s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
:ok_hand: All tests passed!
 
Tests run: **3601**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/40/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :warning: Build finished in 28m 57s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.di.trans.step.mqtt.MQTTStreamSourceTest.testMqttStreamSingleTopic</b> (click to expand)</summary><p>

```
${result.errorDetails}
```

</p></details>

  
Tests run: **3607**, Failures: **1**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/42/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 25m 28s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
:ok_hand: All tests passed!
 
Tests run: **3610**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/43/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 24m 24s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
:ok_hand: All tests passed!
 
Tests run: **3619**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/45/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 24m 50s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
:ok_hand: All tests passed!
 
Tests run: **3619**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/46/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 27m 37s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
:ok_hand: All tests passed!
 
Tests run: **3616**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/47/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 26m 2s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
:ok_hand: All tests passed!
 
Tests run: **3616**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-6517/48/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**